src/components/plots/plot.ts

```diff
@@ -124,7 +108,7 @@ module Plottable {
     }
 
     protected _setupDatasetNodes(dataset: Dataset) {
-      var drawer = this._key2PlotDatasetKey.get(this._keyForDataset(dataset)).drawer;
+      var drawer = this._datasetToDrawer.get(dataset);
```

Comment:
Let's leave it how it currently is, since the separation of functionality is good.